rewrite-core/src/main/java/org/openrewrite/trait/VisitFunction3.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.trait;
+
+import org.openrewrite.Cursor;
+import org.openrewrite.Tree;
+
+public interface VisitFunction3<U extends Trait<?>, P> {
+    Tree visit(U data, Cursor cursor, P p);
```

Comment:
On second thought, `VisitFunction3` is superfluous since `Trait#getCursor` exists. Removing the whole interface in ee129b1d2b8a61ca4e18b35dddcf5c685a45512f.